feat: add blog post about pr-size-labeler (#316)
BedrockStreaming/tech.bedrockstreaming.com: _posts/2022-05-12-github-action-pr-size-labeler.md

---
layout: post
title: Announcing BedrockStreaming/pr-size-labeler github action 🎉
description: We have just published a github action that dynamically adds size labels to _Pull Requests_ based on the size of the diff and the number of modified files.
author: a_caron
lang: en
tags: [oss, github, devops]
color: rgb(251,87,66)
---

![laurel branch with a white paper label](/images/posts/pr-size-labeler/helena-hertz-wWZzXlDpMog-unsplash.jpg)

## Smaller PR for a reduced mental load

For several years at Bedrock Streaming the technical teams have used the _Pull Requests_ code review for each project. 
Between collective ownership, quality improvement, regression detection, knowledge sharing, learning, there is no question in this article to further legitimize the immense interest to implement this practice in your teams.

> This practice can however lead to some problems, each developer who proposes Pull Requests for review by his colleagues can sometimes propose monstrous diffs.

Reviewer: ```suggestion
This practice can however lead to some problems when developers propose Pull Requests for review with monstrous diffs.
```

We are all different after all!

**The size of a dev's PRs does not reflect his productivity at all, it just allows to evaluate the personal and collective mental load produced.**
There are other measures we would like to follow, but let's start with the size of the PR.

Be warned, the purpose of this metric is not to say _"Oh! you made an XL size PR that's not right"_ 😡.
It happens from time to time, and it's not bad.
You should rather look at the distribution of PR sizes of a dev.

Let's take the example of a dev named Bob who would have this distribution over the last month:
![PR size distribution for Bob over the last month.](/images/posts/pr-size-labeler/PR-size-distribution-for-Bob-over-the-last-month.png)

Here we see that Bob is globally making large PRs (taking arbitrary t-shirt sizes), seeing this we can say: _As a TechLead, how can I best accompany Bob to make smaller PRs?_

Next, let's look at Alice's profile, which has a more centered distribution:
![PR size distribution for Alice over the last month.](/images/posts/pr-size-labeler/pr-size-distribution-for-alice-over-the-last-month.png)

> Here, we can say that overall the majority of RPs are of moderate size (in this absolute scale), so the mental load should be lower than for Bob.

Reviewer: ```suggestion
Here, we can say that overall the majority of PRs are of moderate size (in this absolute scale), so the mental load should be lower than for Bob.
```
